tests/unit/test_metrics.py

```diff
@@ -0,0 +1,15 @@
+import sys
+import unittest
+
+from bentoml import metrics
+
+
+class MetricsTestCase(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        # set up api instace so I can test if metric is found in it
+        pass
+
+    def test_metrics_get_lazy_loaded(self):
+        my_his = metrics.Histogram(name="", documentation="some histo")
+        self.assertFalse("prometheus_client" in sys.modules)
```

Comment:
An example test for pytest
```suggestion
from __future__ import annotations

import bentoml
import pytest

@pytest.mark.parameterize("metrics_type", filter(lambda x: isinstance(x, bentoml.metrics.BaseMetric), dir(bentoml.metrics)))
def test_metrics_initialization(metrics_type: t.Type[bentoml.metrics.BaseMetric]):
    m = metrics_type(name="test_metrics", documentation="test")
    assert "prometheus_client" not in sys.modules
```

you can then run it with
```bash
pytest tests/unit -k "test_metrics_initialization"
```

Make sure to install test requirements with `pip install -r requirements/test-requirements.txt`